h2o-core/src/main/java/water/H2ONode.java

```diff
@@ -145,7 +145,16 @@ public boolean isHealthy(long now) {
   // *interned*: there is only one per InetAddress.
   static private H2ONode intern( H2Okey key ) {
     H2ONode h2o = INTERN.get(key);
-    if( h2o != null ) return h2o;
+    if( h2o != null){
+      if(h2o._heartbeat._client && h2o._removed_from_cloud){
+        // the client has reconnected, we need to restore the state
```

Comment:
it would be good to have this in a method (eg. startSendThread) close to the method `stopSendThread`

Reply:
good point, will do!

Reply:
done